xarray/tests/__init__.py

```diff
@@ -182,6 +182,13 @@ def assertDataArrayAllClose(self, ar1, ar2, rtol=1e-05, atol=1e-08):
         assert_allclose(ar1, ar2, rtol=rtol, atol=atol)
 
 
+@contextmanager
+def raises_regex(error, pattern):
+    with pytest.raises(error) as excinfo:
+        yield
+    excinfo.match(pattern)
```

Comment:
Yeah, I'm not opposed to requiring pytest 3 when we have something that really needs it. But the work around here is barely an inconvenience.